src/persist-types/src/lib.rs

```diff
@@ -76,3 +76,10 @@ pub trait Codec64: Sized + 'static {
     /// handle bytes output by all previous versions of encode.
     fn decode(buf: [u8; 8]) -> Self;
 }
+
+/// An opaque fencing token used in compare_and_downgrade_since.
+pub trait Opaque: Clone + Sized + 'static {
```

Comment:
Is `Ord` needed too? Or will that be added in the other PR?

Reply:
persist doesn't care about Ord, actually! we only operate on whether they match or not and the resolution of what to do is left to the persist caller. so, concretely, we can add that to `Epoch`, but don't need it here

Reply:
Does it need `Eq` then?

Reply:
This trait looks great!

Instead of `Eq` bound can I propose a `fn should_exchange(&self, other: &Other) -> bool` method on the trait? It has the same signature as `Eq` but it will allow us to put the epoch logic in one place on the trait and persist can still be oblivious to how we choose whether to exchange or not. That method would be called by persist only when a value is already there and if `should_exchange` return `false` the `compare_and_downgrade` method would with an error.

Reply:
wfm! incoming

Reply:
wait hold up, I think I misunderstood the suggestion. is this also absorbing the `>` logic? so e.g. if expected was 7 and actual was 6, then `should_exchange` would return true and the CaDS would be successful even though the token didn't match exactly?

probably still fine, but I want to think about it some more